icons/solidity.svg

```diff
@@ -0,0 +1 @@
+<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Solidity icon</title><path d="M15.875 10.287l-3.855 6.856H4.305L8.16 24h7.715l3.857-6.857zM8.123 0L4.268 6.857l3.855 6.856 3.856-6.856h7.714L15.838 0z"/></svg>
```

Comment:
As mentioned in https://github.com/simple-icons/simple-icons/pull/4737#discussion_r556490086 the path matches up perfectly to the monochrome from-source variant, but will ping @simple-icons/maintainers to see if one with gaps where the colours change would be preferred.

Reply:
From [this image](https://soliditylang.org/images/SolBlueGradient.png) on their [website](https://soliditylang.org/), it would seem to me that gaps between the differing shades of grey are the way to go.

Reply:
@loganmarchione - would you mind looking to add the gaps to this one too please? 🙏🏻

Reply:
@adamrusted @PeterShaggyNoble 
Here is the shape so you can see the cuts (in red). The only way I could get all the corners to line up was to do two big cuts and one short cut on each half of the shape (for a total of 6 lines).
`<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M15.875 10.287l-3.855 6.856H4.305L8.16 24h7.715l3.857-6.857zM8.123 0L4.268 6.857l3.855 6.856 3.856-6.856h7.714L15.838 0z"/><path d="M4.305 17.143h15.427" fill="none" stroke="red" stroke-width=".5"/><path d="M8.16 24l7.715-13.713" fill="none" stroke="#fe0000" stroke-width=".5"/><path d="M12.02 17.143L15.875 24M8.123 13.713L15.838 0M4.268 6.857h15.425M8.123 0l3.856 6.857" fill="none" stroke="red" stroke-width=".5"/></svg>`

Here is attempt 1. Thoughts?
`<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Solidity icon</title><path d="M8.41 0l3.569 6.346L15.55 0H8.41zM7.98.256l-3.57 6.351h7.144L7.979.256zm8.003 0l-3.574 6.351h7.145L15.983.256zM4.408 7.107l3.57 6.35 3.573-6.35H4.408zm11.612 3.436l-3.575 6.35h7.147l-3.572-6.35zm-11.575 6.85l3.57 6.351 3.575-6.351H4.445zm8.002 0l3.573 6.351 3.572-6.351h-7.145zm-.427.257L8.445 24h7.143l-3.568-6.35z"/></svg>`
![solidity](https://user-images.githubusercontent.com/4955337/104493587-2279db00-55cd-11eb-8c2e-2984ddabfdfa.png)

Reply:
@loganmarchione try the below path. I've moved the inner paths in to match the source file, though it's probably not perfect as I manually moved the points in Figma with the source file underneath it! 😅

```svg
<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Solidity icon</title><path d="M8.4 0L12 6.3 15.5 0H8.4zM8 .3L4.4 6.6h7.1L8 .3zm8 0l-3.8 6.6h7.5L16 .3zM4.4 7.1l3.7 6.6 3.7-6.6H4.4zm11.5 3.2l-3.7 6.6h7.4l-3.7-6.6zM4.3 17.1L8 23.7l3.7-6.6H4.3zm8.1.3l3.6 6.4 3.6-6.4h-7.2zm-.4.3L8.4 24h7.1c.1 0-3.5-6.4-3.5-6.3z"/></svg>
```

Reply:
@adamrusted  - I don't think the meeting points of all the corners in your shape match up like they do in [this file](https://soliditylang.org/images/SolBlueGradient.png).
![Screenshot_20210113_224205](https://user-images.githubusercontent.com/4955337/104541930-7ced5880-561a-11eb-8e7b-99ee9b303465.png)
).

Reply:
@loganmarchione no, this is where we hit a bit of a 'subjective' grey area. I'm trying to match the corners as close as I can to the original SVG on their site, just with the gaps between sections in place. Otherwise you end up with the icon we hold not matching the 'shape' of the original. Happy for @fbernhart or @PeterShaggyNoble to take a look at this though and see which they'd prefer.

Reply:
The version on their website is slightly different, @adamrusted:

![image](https://user-images.githubusercontent.com/15157491/104577311-3cf59800-5651-11eb-8f15-fdb65af8bcfe.png)

Reply:
@PeterShaggyNoble - comparing @loganmarchione's version to the PNG gives the below result, the reason I moved the gaps in was to try and match the [SVG source](https://docs.soliditylang.org/en/v0.8.0/brand-guide.html) as closely as possible, just with the gaps between the colours as I've done with other icons. If we're saying the source is the blue PNG given that contains the gaps already, then the icon will unfortunately need to be rebuilt, as the shape of the icon is fairly different in the PNG.

![image](https://user-images.githubusercontent.com/1830867/104580653-6d3f3580-5655-11eb-80b4-ea11d49adeec.png)

Reply:
Ahhh ok I misunderstood you. I see what you mean now. 

Personally, I think the source should be the SVG (shades of gray with no gaps) since it's the one mentioned in their [brand guide](https://docs.soliditylang.org/en/v0.7.4/brand-guide.html), on their [main page](https://soliditylang.org/), and is the [favicon](https://soliditylang.org/favicon.png). The blue PNG is definitely more visually distinct, but seems to only be used that one time on the home page.

Reply:
I'll export the path I used in the above example, if you want to use that one @loganmarchione?

Reply:
Just pushed.